litestar/middleware/csrf.py

```diff
@@ -111,17 +110,17 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             form = await request.form()
             existing_csrf_token = form.get("_csrf_token", None)
 
+        connection_state = ScopeState.from_scope(scope)
         if request.method in self.config.safe_methods or should_bypass_middleware(
             scope=scope,
             scopes=self.scopes,
             exclude_opt_key=self.config.exclude_from_csrf_key,
             exclude_path_pattern=self.exclude,
         ):
-            token = csrf_cookie or generate_csrf_token(secret=self.config.secret)
-            set_litestar_scope_state(scope=scope, key=SCOPE_STATE_CSRF_TOKEN_KEY, value=token)
+            token = connection_state.csrf_token = csrf_cookie or generate_csrf_token(secret=self.config.secret)
             await self.app(scope, receive, self.create_send_wrapper(send=send, csrf_cookie=csrf_cookie, token=token))
         elif self._csrf_tokens_match(existing_csrf_token, csrf_cookie):
-            set_litestar_scope_state(scope=scope, key=SCOPE_STATE_CSRF_TOKEN_KEY, value=existing_csrf_token)
+            connection_state.csrf_token = existing_csrf_token or ""
```

Comment:
Was this defaulting to an empty string there before as well?

Reply:
No, but here's the current `main` implementation for reference:

https://github.com/litestar-org/litestar/blob/7414f7fd7d4782223502895e6a23b77ed635cd2d/litestar/middleware/csrf.py#L87-L127

Look at line 105, we use `dict.get()` to set the value of `existing_csrf_token` so it can be `None` if the header doesn't exist.

At line 123, that block is guarded by `self._csrf_tokens_match()` which will return `False` if it is `None`, so actually `existing_csrf_token` cannot be falsy in this block, its just that its value is not narrowed appropriately.

So, I should probably use a `cast()` or `type: ignore` instead of the `or` here, but even better would be to appropriately narrow the types, which would probably be as easy as using `request.cookies.get(..., "")` and `request.headers.get(..., "")` on lines 104 and 105 respectively, and re-type the `_csrf_tokens_match()` method to only accept strings.

Reply:
changed to `type: ignore` and opened https://github.com/litestar-org/litestar/issues/2770